internal/provider/telemetry_resource_test.go

```diff
@@ -931,12 +921,187 @@ resource "modtm_telemetry" "test" {
   tags = {
    %s
   }
+}
+`, endpointAssignment, enabledAssignment, sb.String())
+	return r
+}
+
+func testAccTelemetryResourceConfigWithModulePath(endpointAssignment string, enabled bool, tags map[string]string, modulePath string) string {
+	if endpointAssignment != "" {
+		endpointAssignment = fmt.Sprintf("endpoint = \"%s\"", endpointAssignment)
+	}
+	enabledAssignment := ""
+	if !enabled {
+		enabledAssignment = "enabled = false"
+	}
+	sb := strings.Builder{}
+	for k, v := range tags {
+		sb.WriteString(fmt.Sprintf("%s = \"%s\"", k, v))
+		sb.WriteString("\n")
+	}
+	r := fmt.Sprintf(`
+provider "modtm" {
   %s
+  %s
+}
+
+resource "modtm_telemetry" "test" {
+  tags = {
+   %s
+  }
+	module_path = "%s"
 }
-`, endpointAssignment, enabledAssignment, sb.String(), nonceAssignment)
+`, endpointAssignment, enabledAssignment, sb.String(), modulePath)
 	return r
 }
 
-func intPtr(i int) *int {
-	return &i
+func TestModulePathToKey_Success(t *testing.T) {
+	t.Setenv("TF_DATA_DIR", ".terraform")
+	defer os.Unsetenv("TF_DATA_DIR")
+
+	tests := []struct {
+		name       string
+		modulePath string
+		expected   string
+	}{
+		{"Valid module path", ".terraform/modules/abc123", "abc123"},
+		{"Nested module path", ".terraform/modules/nested/abc123", "nested"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			key, err := modulePathToKey(tt.modulePath)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expected, key)
+		})
+	}
+}
+
+func TestModulePathToKey_Failure(t *testing.T) {
+	tests := []struct {
+		name       string
+		modulePath string
+	}{
+		{"Non-matching module path", "/path/to/nonexistent"},
+		{"Empty module path", ""},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := modulePathToKey(tt.modulePath)
+			assert.Error(t, err)
+		})
+	}
+}
+
+func TestModulePathToKey_WithCustomTFDataDir(t *testing.T) {
+	customTFDataDir := "/custom/terraform/data"
+	t.Setenv("TF_DATA_DIR", customTFDataDir)
+	defer os.Unsetenv("TF_DATA_DIR")
```

Comment:
Based on [document](https://beta.pkg.go.dev/testing@master#T.Setenv), I think we can skip `Unsetenv` manually here.